astroplan/core.py

```diff
@@ -162,56 +164,336 @@ def altaz(self, time, target=None, obswl=None):
+        ret1 : `~astropy.time.Time`
+            Time of rise/set
+        '''
+        if prev_next == 'next':
+            times = time + np.linspace(0, 1, N)*u.day
+        else:
+            times = time + np.linspace(-1, 0, N)*u.day
+
+        altitudes = self.altaz(times, target).alt
+        horizon_crossing_limits = self._horiz_cross(times, altitudes, rise_set,
+                                                    horizon,
+                                                    return_limits=True,
+                                                    return_alts=True)
+        return self._two_point_interp(*horizon_crossing_limits, horizon=horizon)
+
+    @u.quantity_input(horizon=u.deg)
+    def calc_rise(self, time, target, which='nearest', horizon=0*u.degree):
+        '''
+        Calculate rise time.
+
+        Compute time of the next/previous/nearest rise of the `target`
+        object, where "rise" is defined as the time when the `target`
+        transitions from altitudes below the `horizon` to above the
+        `horizon`.
 
         Parameters
         ----------
         time : `~astropy.time.Time`
-        """
-        raise NotImplementedError()
+            Time of observation
 
-    def sunset(self, time, **kwargs):
-        """
-        Returns the local sunset time.
+        target : coordinate object (i.e. `~astropy.coordinates.SkyCoord`, `~astroplan.FixedTarget`)
+            Target celestial object
 
-        The default sunset returned is the next one to occur.
+        which : str - can be "next", "previous", or "nearest"; defaults to "nearest"
+            Choose which sunrise relative to the present `time` would you
+            like to calculate
+
+        horizon : `~astropy.units.Quantity` (optional), default = zero degrees
+            Degrees above/below actual horizon to use
+            for calculating rise/set times (i.e.,
+            -6 deg horizon = civil twilight, etc.)
+
+        Returns
+        -------
+        `~astropy.time.Time`
+            Rise time of target
+        '''
+        if which == 'next' or which == 'nearest':
+            next_rise = self._calc_riseset(time, target, 'next', 'rising',
+                                           horizon)
+            if which == 'next':
+                return next_rise
+
+        if which == 'previous' or which == 'nearest':
+            previous_rise = self._calc_riseset(time, target, 'previous',
+                                               'rising', horizon)
+            if which == 'previous':
+                return previous_rise
+
+        if which == 'nearest':
+            if abs(time - previous_rise) < abs(time - next_rise):
+                return previous_rise
+            else:
+                return next_rise
+        raise ValueError('"which" kwarg must be "next", "previous" or '
+                         '"nearest".')
+
+    @u.quantity_input(horizon=u.deg)
+    def calc_set(self, time, target, which='nearest', horizon=0*u.degree):
```

Comment:
I addressed this point in the [notes](https://github.com/bmorris3/astroplan/wiki/Notes#on-risingsetting-methods), but I'm interested in what all @astroplanners/astroplanners-contributors think about this. When I here `set_time`, it sounds to me like it's a setter and it's changing an internal time, so I wanted to avoid the `*_time` naming scheme. `target_set_time ` seems longer and less descriptive.